packages/camera/camera_avfoundation/ios/Classes/CameraPlugin.m

```diff
@@ -148,274 +138,355 @@ - (void)availableCamerasWithCompletion:
-    } else if ([@"getExposureOffsetStepSize" isEqualToString:call.method]) {
-      result(@(0.0));
-    } else if ([@"setExposureOffset" isEqualToString:call.method]) {
-      [_camera setExposureOffsetWithResult:result
-                                    offset:((NSNumber *)call.arguments[@"offset"]).doubleValue];
-    } else if ([@"lockCaptureOrientation" isEqualToString:call.method]) {
-      [_camera lockCaptureOrientationWithResult:result orientation:call.arguments[@"orientation"]];
-    } else if ([@"unlockCaptureOrientation" isEqualToString:call.method]) {
-      [_camera unlockCaptureOrientationWithResult:result];
-    } else if ([@"setFocusMode" isEqualToString:call.method]) {
-      [_camera setFocusModeWithResult:result mode:call.arguments[@"mode"]];
-    } else if ([@"setFocusPoint" isEqualToString:call.method]) {
-      BOOL reset = ((NSNumber *)call.arguments[@"reset"]).boolValue;
-      double x = 0.5;
-      double y = 0.5;
-      if (!reset) {
-        x = ((NSNumber *)call.arguments[@"x"]).doubleValue;
-        y = ((NSNumber *)call.arguments[@"y"]).doubleValue;
+        // Request audio permission on `create` call with `enableAudio` argument instead of the
+        // `prepareForVideoRecording` call. This is because `prepareForVideoRecording` call is
+        // optional, and used as a workaround to fix a missing frame issue on iOS.
+        if (settings.enableAudio) {
+          // Setup audio capture session only if granted audio access.
+          FLTRequestAudioPermissionWithCompletionHandler(^(FlutterError *error) {
+            // cannot use the outter `strongSelf`
+            typeof(self) strongSelf = weakSelf;
+            if (!strongSelf) return;
+            if (error) {
+              completion(nil, error);
+            } else {
+              [strongSelf createCameraOnSessionQueueWithName:cameraName
+                                                    settings:settings
+                                                  completion:completion];
+            }
+          });
+        } else {
+          [strongSelf createCameraOnSessionQueueWithName:cameraName
+                                                settings:settings
+                                              completion:completion];
+        }
       }
-      [_camera setFocusPointWithResult:result x:x y:y];
-    } else if ([@"pausePreview" isEqualToString:call.method]) {
-      [_camera pausePreviewWithResult:result];
-    } else if ([@"resumePreview" isEqualToString:call.method]) {
-      [_camera resumePreviewWithResult:result];
-    } else if ([@"setDescriptionWhileRecording" isEqualToString:call.method]) {
-      [_camera setDescriptionWhileRecording:(call.arguments[@"cameraName"]) result:result];
-    } else if ([@"setImageFileFormat" isEqualToString:call.method]) {
-      NSString *fileFormat = call.arguments[@"fileFormat"];
-      [_camera setImageFileFormat:FCPGetFileFormatFromString(fileFormat)];
```

Comment:
This was never calling `result` (and it doesn't pass it in to the implementation), so that wasn't good :|